testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/preload/PreloadMiniAppViewModel.kt

```diff
@@ -0,0 +1,54 @@
+package com.rakuten.tech.mobile.testapp.ui.display.preload
+
+import androidx.lifecycle.*
+import com.rakuten.tech.mobile.miniapp.MiniApp
+import com.rakuten.tech.mobile.miniapp.MiniAppManifest
+import com.rakuten.tech.mobile.miniapp.MiniAppSdkException
+import com.rakuten.tech.mobile.testapp.ui.settings.AppSettings
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class PreloadMiniAppViewModel constructor(
+    private val miniApp: MiniApp
+) : ViewModel() {
+
+    constructor() : this(MiniApp.instance(AppSettings.instance.miniAppSettings))
+
+    private val _miniAppManifest = MutableLiveData<MiniAppManifest>()
+    private val _miniAppManifestMetadata = MutableLiveData<String>()
+    private val _miniAppVersionId = MutableLiveData<String>()
+    private val _manifestErrorData = MutableLiveData<String>()
+    private val _versionIdErrorData = MutableLiveData<String>()
+
+    val miniAppManifest: LiveData<MiniAppManifest>
+        get() = _miniAppManifest
+    val miniAppManifestMetadata: LiveData<String>
+        get() = _miniAppManifestMetadata
+    val miniAppVersionId: LiveData<String>
+        get() = _miniAppVersionId
+    val manifestErrorData: LiveData<String>
+        get() = _manifestErrorData
+    val versionIdErrorData: LiveData<String>
+        get() = _versionIdErrorData
+
+    fun getMiniAppManifest(miniAppId: String, versionId: String, metadataKey: String) =
+        viewModelScope.launch(Dispatchers.IO) {
+            try {
+                val miniAppManifest = miniApp.getMiniAppManifest(miniAppId, versionId)
+                _miniAppManifest.postValue(miniApp.getMiniAppManifest(miniAppId, versionId))
```

Comment:
Why do you need to call `miniApp.getMiniAppManifest` here again? Can't you just pass `miniAppManifest` from abov?

Reply:
Ah, I don't know why I wrote like this! Sorry, I am updating.